data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/actions/HistogramAggregateAction.java

```diff
@@ -0,0 +1,214 @@
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Map;
+import java.util.List;
+import java.util.HashMap;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Optional;
+
+/**
+ * An AggregateAction that combines multiple Events into a single Event. This action will create a combined event with histogram buckets of the values 
+ * of specified list of keys from the groupState on concludeGroup. 
+ * @since 2.1
+ */
+@DataPrepperPlugin(name = "histogram", pluginType = AggregateAction.class, pluginConfigurationType = HistogramAggregateActionConfig.class)
+public class HistogramAggregateAction implements AggregateAction {
+    private static final String DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSXXX";
+    static final String EVENT_TYPE = "event";
+    public static final String HISTOGRAM_METRIC_NAME = "histogram";
+    public static final String SUM_KEY = "sum";
+    public static final String COUNT_KEY = "count";
+    public static final String BUCKETS_KEY = "buckets";
+    public static final String BUCKET_COUNTS_KEY = "bucket_counts";
+    public static final String MIN_KEY = "min";
+    public static final String MAX_KEY = "max";
+    public static final String START_TIME_KEY = "startTime";
+    public final String countKey;
+    public final String bucketCountsKey;
+    public final String bucketsKey;
+    public final String startTimeKey;
+    public final String outputFormat;
+    public final String sumKey;
+    public final String maxKey;
+    public final String minKey;
+    public final String key;
+    public final String units;
+    public final String keyPrefix;
+    public final boolean recordMinMax;
+
+    private long startTimeNanos;
+    private double[] buckets;
+
+    @DataPrepperPluginConstructor
+    public HistogramAggregateAction(final HistogramAggregateActionConfig histogramAggregateActionConfig) {
+        this.key = histogramAggregateActionConfig.getKey();
+        List<Number> bucketList = histogramAggregateActionConfig.getBuckets();
+        this.buckets = new double[bucketList.size()+2];
+        int bucketIdx = 0;
+        this.buckets[bucketIdx++] = -Float.MAX_VALUE;
+        for (int i = 0; i < bucketList.size(); i++) {
+            this.buckets[bucketIdx++] = convertToDouble(bucketList.get(i));
+        }
+        this.buckets[bucketIdx] = Float.MAX_VALUE;
+        Arrays.sort(this.buckets);
+        this.keyPrefix = histogramAggregateActionConfig.getGeneratedKeyPrefix();
+        this.bucketCountsKey = this.keyPrefix + BUCKET_COUNTS_KEY;
+        this.bucketsKey = this.keyPrefix + BUCKETS_KEY;
+        this.countKey = this.keyPrefix + COUNT_KEY;
+        this.sumKey = this.keyPrefix + SUM_KEY;
+        this.maxKey = this.keyPrefix + MAX_KEY;
+        this.minKey = this.keyPrefix + MIN_KEY;
+        this.startTimeKey = this.keyPrefix + START_TIME_KEY;
```

Comment:
Instead of having all these variables as part of the `AggregateAction`, can we move all of them to the `HistogramAggregateActionConfig`? That way this class would be much cleaner, since it could hold the `HistogramAggregateActionConfig` as its private variable, and use getters that construct the value. For example

```
getSumKey() {
    return this.keyPrefix + SUM_KEY;
}
```